modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -42,22 +42,18 @@ func (k Keeper) OnChanOpenInit(
 		err      error
 		metadata icatypes.Metadata
 	)
-	if strings.TrimSpace(version) == "" {
```

Comment:
I am wondering now if we should just leave this as is. For our implementation of ICA controller it's easy to figure out the host connection ID, so we could just use it. Other implementations can leave it empty, if they cannot retrieve it, and with the change added to `OnChanOpenTry` in the host, then the host connection ID will be set. The change in `OnChanOpenTry` is actually what enables to leave the host connection ID empty in the proposed metadata of the controller. 

What do you think, @srdtrk?

Reply:
Yeah, I think we could do that. Actually, that is my first approach. The only thing is we need to allow the [validation](https://github.com/cosmos/ibc-go/blob/97a52e35f35dadfc08aa9e852196e151eeaa56af/modules/apps/27-interchain-accounts/controller/keeper/handshake.go#L59) on controller chain that allow empty host connectionID